vars/sonarExecuteScan.groovy

```diff
@@ -1,231 +1,88 @@
-                    if(config.projectVersion) config.options.add("sonar.projectVersion=${config.projectVersion}")
-                    // prefix options
-                    config.options = config.options.collect { it.startsWith('-D') ? it : "-D${it}" }
-                    withEnv(envVars){
-                        sh "PATH=\$PATH:'${env.WORKSPACE}/.sonar-scanner/bin' sonar-scanner ${config.options.join(' ')}"
-                    }
+        def utils = parameters.juStabUtils ?: new Utils()
+        stepParameters.remove('juStabUtils')
+
+        def jenkinsUtils = parameters.jenkinsUtilsStub ?: new JenkinsUtils()
+        stepParameters.remove('jenkinsUtilsStub')
+
+        new PiperGoUtils(this, utils).unstashPiperBin()
+        utils.unstash('pipelineConfigAndTests')
+        script.commonPipelineEnvironment.writeToDisk(script)
+
+        writeFile(file: ".pipeline/tmp/${METADATA_FILE}", text: libraryResource(METADATA_FILE))
+
+        withEnv([
+            "PIPER_parametersJSON=${groovy.json.JsonOutput.toJson(stepParameters)}",
+        ]) {
+            // get context configuration
+            Map config = readJSON(text: sh(returnStdout: true, script: "./piper getConfig --contextConfig --stepMetadata '.pipeline/tmp/${METADATA_FILE}'"))
+            echo "Config: ${config}"
+
+            Map stepConfig = readJSON(text: sh(returnStdout: true, script: "./piper getConfig --stepMetadata '.pipeline/tmp/${METADATA_FILE}'"))
+            echo "StepConfig: ${stepConfig}"
+
+            // determine credentials to load
+            List credentials = []
+            if (config.sonarTokenCredentialsId)
+                credentials.add(string(credentialsId: config.sonarTokenCredentialsId, variable: 'PIPER_token'))
+            if(isPullRequest()){
+                checkMandatoryParameter(config, "owner")
+                checkMandatoryParameter(config, "repository")
+                if(config.legacyPRHandling) {
+                    checkMandatoryParameter(config, "githubTokenCredentialsId")
+                    if (config.githubTokenCredentialsId)
+                        credentials.add(string(credentialsId: config.githubTokenCredentialsId, variable: 'PIPER_githubToken'))
                 }
-            } finally {
-                sh 'rm -rf .sonar-scanner .certificates .scannerwork'
             }
-        }
-
-        if(configuration.sonarTokenCredentialsId){
-            def workerForSonarAuth = worker
-            worker = { config ->
-                withCredentials([string(
-                    credentialsId: config.sonarTokenCredentialsId,
-                    variable: 'SONAR_TOKEN'
-                )]){
-                    config.options.add("sonar.login=$SONAR_TOKEN")
-                    workerForSonarAuth(config)
+            // load certificates into cacerts file
+            loadCertificates([customTlsCertificateLinks: stepConfig.customTlsCertificateLinks, verbose: stepConfig.verbose])
```

Comment:
Not 100% sure, but I think you can omit the square brackets: Groovy will convert the key:value pairs to a Map automatically.